pkg/daemon/daemon.go

```diff
@@ -1617,7 +1617,7 @@ func (dn *Daemon) updateSSHKeyLocation(cfg *mcfgv1.MachineConfig) error {
 		return fmt.Errorf("ignition failure when updating SSH key location: %w", err)
 	}
 
-	if err := dn.updateSSHKeys(ignConfig.Passwd.Users); err != nil {
+	if err := dn.updateSSHKeys(ignConfig.Passwd.Users, ignConfig.Passwd.Users); err != nil {
```

Comment:
Is this somehow leftover from your work around SSH? It seems out of place for this PR.